src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/diagnostics/Diagnostics.java

```diff
@@ -0,0 +1,239 @@
+import com.sonymobile.tools.gerrit.gerritevents.dto.GerritChangeKind;
+import com.sonymobile.tools.gerrit.gerritevents.dto.attr.Account;
+import com.sonymobile.tools.gerrit.gerritevents.dto.attr.Change;
+import com.sonymobile.tools.gerrit.gerritevents.dto.attr.PatchSet;
+import hudson.Main;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+import jenkins.model.ModelObjectWithChildren;
+import jenkins.model.ModelObjectWithContextMenu;
+import org.acegisecurity.Authentication;
+import org.apache.commons.lang3.StringUtils;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Sub page on {@link com.sonyericsson.hudson.plugins.gerrit.trigger.GerritManagement} containing some diagnostic views.
+ */
+public class Diagnostics implements ModelObjectWithChildren, ModelObjectWithContextMenu {
+
+    @Override
+    public ContextMenu doChildrenContextMenu(StaplerRequest request, StaplerResponse response) throws Exception {
+        return getContextMenu(null);
+    }
+
+    //CS IGNORE LineLength FOR NEXT 8 LINES. REASON: Javadoc
+    /**
+     * Helper method to produce the breadcrumb context menu.
+     *
+     * @param context the url prefix to put on all urls.
+     * @return the selectable reports.
+     * @see #doChildrenContextMenu(StaplerRequest, StaplerResponse)
+     * @see com.sonyericsson.hudson.plugins.gerrit.trigger.GerritManagement#doContextMenu(StaplerRequest, StaplerResponse)
+     */
+    @Restricted(NoExternalUse.class)
+    public ContextMenu getContextMenu(String context) {
+        ContextMenu menu = new ContextMenu();
+        String url = makeRelativeUrl(context, "buildMemory");
+        menu.add(new MenuItem()
+                         .withUrl(url)
+                         .withStockIcon("clipboard.png")
+                         .withDisplayName(Messages.BuildMemoryReport_DisplayName()));
+        url = makeRelativeUrl(context, "eventListeners");
+        menu.add(new MenuItem()
+                         .withUrl(url)
+                         .withStockIcon("clipboard.png")
+                         .withDisplayName(Messages.EventListenersReport_DisplayName()));
+        if (isDebugMode()) {
+            menu.add("triggerDebugEvent", "warning.png", "Trigger Debug", false, true);
+        }
+        return menu;
+    }
+
+    /**
+     * Construct an url with a potential context for {@link #getContextMenu(String)}.
+     * @param context potential string to prefix with
+     * @param name the url
+     * @return an url, relative or not.
+     */
+    private String makeRelativeUrl(String context, String name) {
```

Comment:
java.net.URI seems to require absolute url with schema and all.
Jenkins has a `Functions.joinPath(String... components)` but it doesn't handle `null` values for path components so that if statement would need to be here in this case anyway.